core/tests/unittests/test_feature_selection.py

```diff
@@ -0,0 +1,89 @@
+from autogluon.core.utils.feature_selection import *
```

Comment:
should we consider a more end-to-end test of feature-selection on a tiny toy dataset (verifying that the irrelevant feature is actually removed)? in particular is there any way to implement such a test that it can run in a few seconds?

Reply:
one edge case is dataset with only 1 feature, have you verified this is ok there?

Reply:
We didn't previously feature prune in that edge case (because model fit throws an exception) but I added an if statement + logs to make things more clear. Also because we are pruning features based on a noise column's feature importance + validation performance increase, there are no guarantees that all irrelevant features will be removed (although a good number of them will probably be pruned). One option is displaying a warning if no irrelevant features are removed. I'll get to this if I have time after doing everything else.